crawl-ref/source/dgn-overview.h

```diff
@@ -37,12 +37,15 @@ string get_level_annotation(level_id li = level_id::current(),
 
 bool level_annotation_has(string str, level_id li = level_id::current());
 
-void annotate_level();
-void do_annotate(level_id& li);
 void clear_level_annotations(level_id li);
 
 class writer;
 class reader;
 void marshallUniqueAnnotations(writer& outf);
 void unmarshallUniqueAnnotations(reader& inf);
 bool connected_branch_can_exist(branch_type br);
+
+// XXX I botched this in to get crawl to compile, but don't actually understand
+// on a deep level what it does.
+typedef map<level_id, string> annotation_map_type;
```

Comment:
This line defines a type called `annotation_map_type`, which will be available to every file that includes `dgn-overview.h`. Currently only `dgn-overview.cc` uses this type, so adding this line is not a good idea; ideally, we'd like to keep the data structure used to store annotations (the `map<level_id, string>` bit, local to the individual file, not every file that includes the corresponding header.

The next line declares a variable called `level_annotations` located "somewhere else"; again, since this is in `dgn-overview.h`, this is accessible to every file that includes this header, and any file can use this global variable directly. Again, including this line is not a good idea because it exposes the internal data structure unnecessarily.

Instead, these definitions should go at the top of `annotate.cc`, where you've added them commented out.